File: src/mod10_design_pattern/lesson12_strategy/java/Main.java
package mod10_design_pattern.lesson12_strategy.java;

public class Main {
    public static void main(String[] args) {

        StudyStrategy bookStrategy = new BookStudyStrategy();
        Student hermione = new Student("Hermione Granger", bookStrategy);
        hermione.learnSpell("Vingardium Levioza");

        hermione.setStudyStrategy(new VideoStudyStrategy());
        hermione.learnSpell("Expelliarmus");

        Student harry = new Student("Harry Potter", new PracticalStudyStrategy());
        harry.learnSpell("Riktumsempra");


    }
}

File: src/mod10_design_pattern/lesson12_strategy/kotlin/Main.kt
package mod10_design_pattern.lesson12_strategy.kotlin

fun main() {

    val bookStrategy: StudyStrategy = BookStudyStrategy()
    val hermione = Student("Hermione Granger", bookStrategy)
    hermione.learnSpell("Vingardium Levioza")

    hermione.setStudyStrategy(VideoStudyStrategy())
    hermione.learnSpell("Expelliarmus")

    val harry = Student("Harry Potter", PracticalStudyStrategy())
    harry.learnSpell("Riktumsempra")
}